prometheus/src/main/scala/filodb/prometheus/ast/Vectors.scala

```diff
@@ -139,10 +139,8 @@ sealed trait Vector extends Expression {
   val regexColumnName: String = "::(?=[^::]+$)" //regex pattern to extract ::columnName at the end
 
   // Convert metricName{labels} -> {labels, __name__="metricName"} so it's uniform
-  lazy val mergeNameToLabels: Seq[LabelMatch] = {
+  val mergeNameToLabels: Seq[LabelMatch] = {
     val nameLabel = labelSelection.find(_.label == PromMetricLabel)
-    if (nameLabel.isEmpty && metricName.isEmpty)
-      throw new IllegalArgumentException("Metric name is not present")
     if (metricName.nonEmpty) {
       if (nameLabel.nonEmpty) throw new IllegalArgumentException("Metric name should not be set twice")
       // metric name specified but no __name__ label.  Add it
```

Comment:
Now, an exception is always thrown during construction if the metric name is set twice.

Reply:
Also, the FiloDB constraint-- that a metric name is present-- is not required at construction.